aten/src/ATen/native/cudnn/LossCTC.cpp

```diff
@@ -68,9 +68,10 @@ bool _use_cudnn_ctc_loss(
   return use_cudnn;
 }
 
-std::tuple<Tensor, Tensor> _cudnn_ctc_loss(const Tensor& log_probs_t, const Tensor& targets_t, IntArrayRef input_lengths_, IntArrayRef target_lengths_, int64_t BLANK, bool deterministic, bool zero_infinity) {
+std::tuple<Tensor, Tensor> _cudnn_ctc_loss(const Tensor& log_probs_t_, const Tensor& targets_t, IntArrayRef input_lengths_, IntArrayRef target_lengths_, int64_t BLANK, bool deterministic, bool zero_infinity) {
   (void)zero_infinity; // only used for backward
   const CheckedFrom c = "cudnn_ctc_loss";
+  Tensor log_probs_t = log_probs_t_.dim() == 3 ? log_probs_t_ : log_probs_t_.unsqueeze(1);
```

Comment:
Can you explain a bit what's going with the cudnn version here? It looks like an unsqueeze is being done but there is no matching resqueeze for the output - am I missing something?

Reply:
Thanks for the catch! Added a test that catches this and fixed the bug